File: learn/perl/var.pl
#!/usr/bin/perl
# -------------------------------------
#   File Name:	var.pl
#      Author:	Zhao Yanbai
#             	Mon Sep  6 10:31:25 2010
# Description:	none
# -------------------------------------

print "hello " x 3;
print "\n";

@array = qw /aaa bbb ccc ddd eee fff/;
foreach $var (@array){
	print $var . " ";
}
print "\n" . $array[2] . " " . $array[$#array] . "\n";
reverse @array;
print "@array";
print "\n";
@array = reverse @array;
print "@array";
print "\n";

push @array, "push_a";
push @array, "push_b";
print "@array";
print "\n";
$var = pop @array;
print "var: $var\n";
print "@array";
print "\n";

unshift @array, "1111";
unshift @array, "****";
print "@array";
print "\n";
$var = shift @array;
print "var: $var\n";
print "@array";
print "\n";



@num = qw !1 5 7 3 6 0 2 8!;
@num = sort @num;
print "@num";
print "\n";


$var_a = "AAA";
$var_b = "BBB";
print "var_a: $var_a\tvar_b:$var_b\n";
($var_a, $var_b) = ($var_b, $var_a);
print "var_a: $var_a\tvar_b:$var_b\n";

File: learn/go/var.go
/*
 * ------------------------------------------------------------------------
 *   File Name: var.go
 *      Author: Zhao Yanbai
 *              2015-11-28 13:47:31 Saturday CST
 * Description: none
 * ------------------------------------------------------------------------
 */
package main

import (
	"fmt"
	"os"
)

func main() {
	defer fmt.Println("Program Exited...")

	var HOME = os.Getenv("HOME")

	fmt.Println(HOME)

	data := int64(0x11223Ace45566)

	// [1] 是使用第1个参数 # 是指要打印出 0 0x 0X等前缀
	fmt.Printf("%d %[1]x %#[1]x %#[1]X\n", data);

}
